### PR TITLE
SG-30560 Code Cleanup: Removing Qt4/PySide/PyQt4 compatibility in Toolkit

### PR DESCRIPTION
**Pull request by eduardoChaucaGallegos**
We are dropping the support and compatibility with Qt4. So we are free to delete all the code in Toolkit/Desktop which supports PySide and PyQt4.

### CODE REVIEW

File: python/tank/authentication/ui/qt5_like_line_edit.py [hunk 1]
```diff
@@ -13,84 +13,7 @@
...
-                p.setPen(col)
-                line_rect.adjust(min_left_bearing, 0, 0, 0)
-
-                elided_text = fm.elidedText(self.placeholderText(), QtCore.Qt.ElideRight, line_rect.width())
-                p.drawText(line_rect, vertical_alignment, elided_text)
-                p.setPen(oldpen)
-else:
-    # Qt5 always has the placeholder.
-    Qt5LikeLineEdit = QtGui.QLineEdit
+Qt5LikeLineEdit = QtGui.QLineEdit
```
Comment (julien-lang) [inline python/tank/authentication/ui/qt5_like_line_edit.py:19]: Is it worth keeping this class or can we switch the code to use QLineEdit?

This one is only in the tk-core authentication module so the change are limited (no other components to consider AFAIK)

File: python/tank/util/qt_importer.py [hunk 2]
```diff
@@ -459,7 +418,7 @@ def _import_modules(self, interface_version_requested):
                 return pyside2
             except ImportError:
                 pass
-        elif interface_version_requested == self.QT5:
+        if interface_version_requested == self.QT5:
```
Comment (julien-lang) [inline python/tank/util/qt_importer.py:421]: Why this change? It looks wrong to me but maybe I'm missing something.

File: tests/util_tests/test_qt_importer.py [hunk 3]
```diff
@@ -160,7 +157,6 @@ def test_qt_importer_with_pyside6_interface_qt4(self):
         assert qt.base["__version__"] is qt.binding_version
 
     @skip_if_pyside6(found=False)
-    @skip_if_pyqt4(found=True)
     @skip_if_pyside(found=True)
```
Comment (julien-lang) [inline tests/util_tests/test_qt_importer.py:160]: What about this one? Should we remove it too?

**Review by julien-lang — COMMENTED** [review]
A good start. I love when we remove code :) Thanks @eduardoChaucaGallegos

File: python/tank/util/qt_importer.py (hunk 2 above)
Comment (eduardoChaucaGallegos) [reply python/tank/util/qt_importer.py:421]: you are right, that was because I was testing commenting the first condition block, but "elif" is the correct one

File: tests/util_tests/test_qt_importer.py (hunk 3 above)
Comment (eduardoChaucaGallegos) [reply tests/util_tests/test_qt_importer.py:160]: yeah, we should. I'm still working on tests

File: python/tank/platform/engine.py [hunk 4]
```diff
@@ -2226,12 +2214,9 @@ def _initialize_dark_look_and_feel(self):
         if self.has_qt5:
             self.log_debug("Applying Qt5-specific styling...")
             self.__initialize_dark_look_and_feel_qt5()
-        elif self.has_qt4:
```
Comment (carlos-villavicencio-adsk) [inline python/tank/platform/engine.py:2229]: Should we need to add another condition for qt6? I don't have context here but just writing this to double check.

File: python/tank/util/qt_importer.py [hunk 5]
```diff
@@ -163,8 +163,6 @@ def _import_module_by_name(self, parent_module_name, module_name):
             logger.debug("Unable to import module '%s': %s", module_name, e)
             pass
         return module
-
-    def _import_pyside(self):
```
Comment (carlos-villavicencio-adsk) [inline python/tank/util/qt_importer.py:167]: I think this line was removed by accident. Can you double check?

File: python/tank/util/qt_importer.py [hunk 6]
```diff
@@ -459,7 +418,7 @@ def _import_modules(self, interface_version_requested):
                 return pyside2
             except ImportError:
                 pass
-        elif interface_version_requested == self.QT5:
```
Comment (carlos-villavicencio-adsk) [inline python/tank/util/qt_importer.py:462]: Can you check if the block before can be deleted? (the one that starts with)

```py
if interface_version_requested == self.QT4:
```

File: tests/util_tests/test_qt_importer.py [hunk 7]
```diff
@@ -107,7 +106,6 @@ def test_qt_importer_with_pyside_interface_qt4(self):
         assert qt.base["__name__"] is qt.binding_name
         assert qt.base["__version__"] is qt.binding_version
 
-    @skip_if_pyqt4(found=False)
```
Comment (carlos-villavicencio-adsk) [inline tests/util_tests/test_qt_importer.py:110]: This tests seems to fail on CI. As the description says:

>         This test only runs if PyQt4 is available and PySide, PySide2 are not available.

Can we consider removing this unit tests entirely?

**Review by carlos-villavicencio-adsk — COMMENTED** [review]
Looks good so far. I just pointed out some comments here and there.

File: python/tank/util/qt_importer.py
Comment (carlos-villavicencio-adsk) [inline python/tank/util/qt_importer.py:1]: Important question in constructor for class `QtImporter`. Is it possible to set a default to `QT5` on the `interface_version_requested` argument?

Comment (julien-lang) [reply python/tank/util/qt_importer.py:1]: No, it's not possible for the moment @carlos-villavicencio-adsk. This is a bigger change that will be addressed in SG-35326

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py
Comment (julien-lang) [inline python/tank/authentication/sso_saml2/core/sso_saml2_core.py:1]: I recommend reviewing this file change in "hide whitespace" mode.

File: python/tank/util/qt_importer.py (hunk 5 above)
Comment (julien-lang) [reply python/tank/util/qt_importer.py:167]: No, actually, this line should be removed along with the rest of the function. We don't want to import pyside anymore.

Comment (carlos-villavicencio-adsk) [reply python/tank/util/qt_importer.py:167]: But the rest of the function doesn't seem deleted, am I missing something? @eduardoChaucaGallegos

Comment (eduardoChaucaGallegos) [reply python/tank/util/qt_importer.py:167]: you are right, rest of the function was not deleted and not given an error because that code was after a return statement as part of function "_import_module_by_name", but now it is deleted

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py [hunk 8]
```diff
@@ -812,7 +669,7 @@ def on_renew_sso_session(self):
         # have been cleared/updated before.
         url = get_renew_path(self._session, self._logger)
         self._logger.debug("Navigating to %s", url)
-        self._view.page().mainFrame().load(url)
+        self._view.page().load(url)
```
Comment (julien-lang) [inline python/tank/authentication/sso_saml2/core/sso_saml2_core.py:672]: What's this change about? It does not look right?

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py [hunk 9]
```diff
@@ -838,7 +695,7 @@ def _polyfill(self):
         environment and define functions which would be required by that code.
         """
         frame = self._view.page().currentFrame()
-        frame.evaluateJavaScript(FUNCTION_PROTOTYPE_BIND_POLYFILL)
+        frame.runJavascript(FUNCTION_PROTOTYPE_BIND_POLYFILL)
```
Comment (julien-lang) [inline python/tank/authentication/sso_saml2/core/sso_saml2_core.py:698]: What's this change about? It does not look right?

File: tests/util_tests/test_qt_importer.py [hunk 10]
```diff
@@ -81,87 +54,7 @@ def test_qt_importer_with_pyside2_interface_qt5(self):
...
-        assert qt.binding_name == "PyQt4"
-        assert qt.base
-        assert qt.base["__name__"] is qt.binding_name
-        assert qt.base["__version__"] is qt.binding_version
-
-    @skip_if_pyside6(found=False)
-    @skip_if_pyqt4(found=True)
-    @skip_if_pyside(found=True)
-    @skip_if_pyside2(found=True)
-    def test_qt_importer_with_pyside6_interface_qt4(self):
```
Comment (julien-lang) [inline tests/util_tests/test_qt_importer.py:138]: No, we don't want to remove that. We want to keep "internal align with PySide"

**Review by julien-lang — CHANGES_REQUESTED** [review]

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py (hunk 8 above)
Comment (julien-lang) [reply python/tank/authentication/sso_saml2/core/sso_saml2_core.py:672]: Nevermind, I got it. This function was a Qt version agnostic alias.

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py (hunk 9 above)
Comment (julien-lang) [reply python/tank/authentication/sso_saml2/core/sso_saml2_core.py:698]: Nevermind, I got it. This function was a Qt version agnostic alias.

File: tests/util_tests/test_qt_importer.py [hunk 11]
```diff
@@ -12,42 +12,15 @@
...
-    skip_if_pyqt4,
 )
 from tank.util import qt_importer
 
 
 class QtImporterTests(TankTestBase):
     """Tests QtImporter functionality."""
 
-    @skip_if_pyside2(found=False)
-    def test_qt_importer_with_pyside2_interface_qt4(self):
```
Comment (julien-lang) [inline tests/util_tests/test_qt_importer.py:27]: We still want this test!

**Review by julien-lang — CHANGES_REQUESTED** [review]

**Review by julien-lang — APPROVED** [review]
LGTM!

File: tests/util_tests/test_qt_importer.py [hunk 12]
```diff
@@ -81,65 +79,13 @@ def test_qt_importer_with_pyside2_interface_qt5(self):
...
     @skip_if_pyside6(found=False)
-    @skip_if_pyqt4(found=True)
-    @skip_if_pyside(found=True)
     @skip_if_pyside2(found=True)
     def test_qt_importer_with_pyside6_interface_qt4(self):
         """
         Test the QtImporter constructor with default interface version requested.
 
-        This test only runs if PySide6 is available and PyQt4, PySide, PySide2 are not available.
+        This test only runs if PySide6 is available and PySide2 are not available.
```
Comment (carlos-villavicencio-adsk) [inline tests/util_tests/test_qt_importer.py:88]:
```suggestion
        This test only runs if PySide6 is available and PySide2 is not available.
```

**Review by carlos-villavicencio-adsk — APPROVED** [review]
Minimal suggestion. All good on my side.

File: python/tank/platform/engine.py [hunk 13]
```diff
@@ -2227,12 +2215,9 @@ def _initialize_dark_look_and_feel(self):
         if self.has_qt5 or self.has_qt6:
             self.log_debug("Applying Qt5/Qt6-specific styling (Fusion)...")
             self.__initialize_dark_look_and_feel_qt5_qt6()
-        elif self.has_qt4:
-            self.log_debug("Applying Qt4-specific styling...")
-            self.__initialize_dark_look_and_feel_qt4()
         else:
             self.log_warning(
-                "Neither Qt4 or Qt5 is available. Toolkit styling will not be applied."
+                "Neither Qt5 or Qt6 is not available. Toolkit styling will not be applied."
```
Comment (julien-lang) [inline python/tank/platform/engine.py:2220]:
```suggestion
                "Neither Qt5 or Qt6 is available. Toolkit styling will not be applied."
```

**Review by julien-lang — APPROVED** [review]
LGTM!
